### ISSUE TITLE
Changing tomcat user and group in compose.env has no effect

### ISSUE THREAD

**Issue by cbcunc** [Unidata/thredds-docker#206]
Host user and group run as 1000 and 1000 even after changing them to another user and group with permissions in compose.env.

**Comment by julienchastang** [comment]
@cbcunc Could we see an example or sample of the `docker-compose.yml` in question? Thxs.

**Comment by julienchastang** [comment]
and also the accompanying `compose.env`.

**Comment by cbcunc** [comment]
Very little change from stock. Just trying to get it going. The permissions error on files/tomcat-users.xml is due to getting owned by 1000 when the container was started:

```
[thredds@tds2-hydroshare thredds-docker]$ git diff
error: open("files/tomcat-users.xml"): Permission denied
fatal: cannot hash files/tomcat-users.xml
diff --git a/compose.env b/compose.env
index 1084d6d..b6a07d0 100644
--- a/compose.env
+++ b/compose.env
@@ -24,6 +24,6 @@ TDM_XMS_SIZE=1G

 # See https://github.com/Unidata/tomcat-docker#configurable-tomcat-uid-and-gid

-TOMCAT_USER_ID=1000
+TOMCAT_USER_ID=1002

-TOMCAT_GROUP_ID=1000
+TOMCAT_GROUP_ID=1006
diff --git a/docker-compose.yml b/docker-compose.yml
index 29a5159..67f887f 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -16,10 +16,10 @@ services:
       - "8443:8443"
     container_name: thredds
     volumes:
-      - /path/to/your/tomcat/logs/:/usr/local/tomcat/logs/
-      - /path/to/your/thredds/logs/:/usr/local/tomcat/content/thredds/logs/
-      - /path/to/your/tomcat-users.xml:/usr/local/tomcat/conf/tomcat-users.xml
-      - /path/to/your/thredds/directory:/usr/local/tomcat/content/thredds
+      - /var/thredds/tomcat/logs/:/usr/local/tomcat/logs/
+      - /var/thredds/tomcat/content/thredds/logs/:/usr/local/tomcat/content/thredds/l
+      - /home/thredds/thredds-docker/files/tomcat-users.xml:/usr/local/tomcat/conf/to
+      - /var/thredds/tomcat/content/thredds:/usr/local/tomcat/content/thredds
       - /path/to/your/data/directory1:/path/to/your/data/directory1
       - /path/to/your/data/directory2:/path/to/your/data/directory2
     env_file:
```

**Comment by julienchastang** [comment]
Out of curiosity, why `1002`/`1006` for the UID/GID? Is that the UID/GID of the user running the container (i.e., running `docker-compose`)? A good choice would be the UID/GID of the user running the container.  For example,

```bash
grep $USER /etc/passwd
ubuntu:x:1000:1000:Ubuntu:/home/ubuntu:/bin/bash
```

So in this case, a good choice for the UID/GID would be `1000`/`1000`. 

Remember, this UID/GID configurability option  is to ensure that file permissions written by the container are under the control of the user running the container. Nothing more.

**Comment by cbcunc** [comment]
Why? Because 1000:1000 is a user on the VM infrastructure on which I'm running this container which has another purpose. It is in LDAP, it doesn't have a shell, and it is and beyond my control. 1000:1000 is a good choice for you, but not for everybody or even most. It is customary to set the the user in the container to something with meaning on the host. When I install thredds directly without a container, I specify the running user which is the owner of the files and that is an expectation of running thredds.

The user running the container is the same 1002:1006. That's pretty standard. I know UID/GID configurability option is to ensure that the permissions of files written by the container are under the control of the user running the container. And that is precisely what I was trying to do with it, other than trying to evaluate the suitability of this container for running thredds in my organization.

**Comment by julienchastang** [comment]
Hi Chris. Yes, as I mentioned `1000`:`1000` was just an example. In your situation it may be different. At any rate, beyond that, I am not sure what the problem is and it is difficult to diagnose this problem remotely. Here are some things to try:

- Is Docker up-to-date on your host?
- Are you running an Linux distro and version that is supported by Docker?
- The Docker daemon runs as a privileged user, normally. Is that the case on your system?
- Do you have security settings that may be getting in the way? Maybe SELinux?

I am also curious to know if you are running Docker on bare metal Linux or a cloud VM? It should work on both but I have had better luck with the latter.

Also, earlier the `git diff` yields a `Permission denied`. Who owns that  `files/tomcat-users.xml` file?

**Comment by cbcunc** [comment]
1) Docker 18.03.1-ce, the latest stable release.
2) CentOS Linux release 7.5.1804 (Core), which definitely supports Docker
3) Docker daemon runs under systemctl, so yes, privileged.
4) SELinux status: disabled
5) Running on a VMWare VM in a private cloud, just like dozens of other containers I run

It shouldn't be necessary to diagnose this remotely. If it's a problem, it's should be repeatable. So are you able to effectively set the UID/GID using compose.env? If you are, then I probably don't have an issue. It appears to me that the UID/GID in compose.env for the thredds container is not percolating up to the tomcat layer. I say this because I can run docker-compose on a clone of the unidata/tomcat-docker repo with 1002:1006 in the compose.env and it works just fine. The volumes are created with 1002:1006.

**Comment by julienchastang** [comment]
Thanks for the background info. That's helpful. I just realized something: can you ensure you are running `unidata/thredds-docker:latest`. Please verify this by running: `docker pull unidata/thredds-docker:latest` and ensure the `docker-compose.yml` refers to `unidata/thredds-docker:latest`. I think that may have been the problem all along.

**Comment by cbcunc** [comment]
That worked. That's all it was. Thanks so much. I'm submitting a PR to this repo today for the fix to docker-compose.yml.

**Comment by julienchastang** [comment]
Thanks for hanging in there. Indeed the documentation does not work as it stands b/c for the container version mismatch as you found. Looking forward to the PR.

**Comment by cbcunc** [comment]
#207. Build passed. Contributor Agreement signed.

**Comment by cbcunc** [comment]
Merged in #207. Thanks, @julienchastang.
